MCMC/utils.py

```diff
@@ -1,45 +1,46 @@
 import freud
 import numpy as np
+import gsd.hoomd
 from numba import jit
 
-def avg_nn(sim_obj, n_frames, r_max=3):
-    box = [sim_obj.L, sim_obj.L,0]
-    frame_avg_nn = []
-    for points in sim_obj.system_history[-n_frames:]:
-        points = np.append(
-            points, np.zeros((points.shape[0], 1)),axis=1
-        )
+def avg_nn(gsd_file, frame=-1, r_max=3):
+    with gsd.hoomd.open(gsd_file) as f:
+        snap = f[frame]
+        box = snap.configuration.box
+        points = snap.particles.position
+        frame_avg_nn = []
         aq = freud.locality.AABBQuery(box, points)
         nlist = aq.query(points, {'r_max': 3}).toNeighborList()
         neighbors = []
-        for i in range(sim_obj.n_particles):
+        for i in range(snap.particles.N):
             neighbors.append(np.where(nlist[:, 0] == i)[0].shape[0]-1)
             frame_avg_nn.append(np.average(neighbors))
     return np.average(frame_avg_nn)
 
-def structure_factor(sim_obj, n_frames, num_k_values=100, k_max=10):
-    box = [sim_obj.L, sim_obj.L,0]
-    sf = freud.diffraction.StaticStructureFactorDebye(
-            num_k_values=num_k_values, k_max=k_max
-            )
-    for points in sim_obj.system_history[-n_frames:]:
-        points = np.append(
-            points, np.zeros((points.shape[0], 1)),axis=1
-        )
+def structure_factor(gsd_file, start=0, stop=-1, num_k_values=100, k_max=10):
+    with gsd.hoomd.open(gsd_file, "rb") as f:
+        snap = f[0]
```

Comment:
I think we can remove `snap = f[0]`. I don't see that we're using it anywhere before the for loop is called. Once the for loop is called, the `snap` variable is being changed.